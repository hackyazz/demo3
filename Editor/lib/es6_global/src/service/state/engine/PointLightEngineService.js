

import * as PointLightAPI$Wonderjs from "../../../../../../node_modules/wonder.js/lib/es6_global/src/api/light/PointLightAPI.js";
import * as CountLightService$Wonderjs from "../../../../../../node_modules/wonder.js/lib/es6_global/src/service/record/all/light/CountLightService.js";
import * as BufferPointLightService$Wonderjs from "../../../../../../node_modules/wonder.js/lib/es6_global/src/service/record/main/light/point/BufferPointLightService.js";
import * as RecordPointLightMainService$Wonderjs from "../../../../../../node_modules/wonder.js/lib/es6_global/src/service/state/main/light/point/RecordPointLightMainService.js";

function setPointLightColor(value, component, state) {
  return PointLightAPI$Wonderjs.setPointLightColor(component, value, state);
}

function setPointLightIntensity(value, component, state) {
  return PointLightAPI$Wonderjs.setPointLightIntensity(component, value, state);
}

function setPointLightConstant(value, component, state) {
  return PointLightAPI$Wonderjs.setPointLightConstant(component, value, state);
}

function setPointLightLinear(value, component, state) {
  return PointLightAPI$Wonderjs.setPointLightLinear(component, value, state);
}

function setPointLightQuadratic(value, component, state) {
  return PointLightAPI$Wonderjs.setPointLightQuadratic(component, value, state);
}

function setPointLightRange(value, component, state) {
  return PointLightAPI$Wonderjs.setPointLightRange(component, value, state);
}

function getCurrentCount(engineState) {
  return CountLightService$Wonderjs.getLightCount(RecordPointLightMainService$Wonderjs.getRecord(engineState)[/* renderLightArr */8]);
}

function isMaxCountByCount(count, engineState) {
  return (CountLightService$Wonderjs.getLightCount(RecordPointLightMainService$Wonderjs.getRecord(engineState)[/* renderLightArr */8]) + count | 0) > BufferPointLightService$Wonderjs.getBufferMaxCount(/* () */0);
}

function getLightCount(engineState) {
  return CountLightService$Wonderjs.getLightCount(RecordPointLightMainService$Wonderjs.getRecord(engineState)[/* renderLightArr */8]);
}

var create = PointLightAPI$Wonderjs.createPointLight;

var getPointLightGameObject = PointLightAPI$Wonderjs.unsafeGetPointLightGameObject;

var getPointLightColor = PointLightAPI$Wonderjs.getPointLightColor;

var getPointLightIntensity = PointLightAPI$Wonderjs.getPointLightIntensity;

var getPointLightConstant = PointLightAPI$Wonderjs.getPointLightConstant;

var getPointLightLinear = PointLightAPI$Wonderjs.getPointLightLinear;

var getPointLightQuadratic = PointLightAPI$Wonderjs.getPointLightQuadratic;

var getPointLightRange = PointLightAPI$Wonderjs.getPointLightRange;

var setPointLightRangeLevel = PointLightAPI$Wonderjs.setPointLightRangeLevel;

var isMaxCount = PointLightAPI$Wonderjs.isMaxCount;

var setPointLightIsRender = PointLightAPI$Wonderjs.setPointLightIsRender;

var getBufferMaxCount = BufferPointLightService$Wonderjs.getBufferMaxCount;

export {
  create ,
  getPointLightGameObject ,
  getPointLightColor ,
  setPointLightColor ,
  getPointLightIntensity ,
  setPointLightIntensity ,
  getPointLightConstant ,
  setPointLightConstant ,
  getPointLightLinear ,
  setPointLightLinear ,
  getPointLightQuadratic ,
  setPointLightQuadratic ,
  getPointLightRange ,
  setPointLightRange ,
  setPointLightRangeLevel ,
  isMaxCount ,
  getCurrentCount ,
  isMaxCountByCount ,
  setPointLightIsRender ,
  getBufferMaxCount ,
  getLightCount ,
  
}
/* PointLightAPI-Wonderjs Not a pure module */
