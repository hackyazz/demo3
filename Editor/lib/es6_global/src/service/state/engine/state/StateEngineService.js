

import * as StateAPI$Wonderjs from "../../../../../../../node_modules/wonder.js/lib/es6_global/src/api/StateAPI.js";

var deepCopyForRestore = StateAPI$Wonderjs.deepCopyForRestore;

var createState = StateAPI$Wonderjs.createState;

var createStateData = StateAPI$Wonderjs.createStateData;

var getStateFromData = StateAPI$Wonderjs.getStateFromData;

var unsafeGetState = StateAPI$Wonderjs.unsafeGetState;

var setStateToData = StateAPI$Wonderjs.setStateToData;

var setState = StateAPI$Wonderjs.setState;

var setIsDebug = StateAPI$Wonderjs.setIsDebug;

export {
  deepCopyForRestore ,
  createState ,
  createStateData ,
  getStateFromData ,
  unsafeGetState ,
  setStateToData ,
  setState ,
  setIsDebug ,
  
}
/* StateAPI-Wonderjs Not a pure module */
