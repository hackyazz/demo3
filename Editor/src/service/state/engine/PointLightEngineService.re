open Wonderjs;

let create = PointLightAPI.createPointLight;

let getPointLightGameObject = PointLightAPI.unsafeGetPointLightGameObject;

let getPointLightColor = PointLightAPI.getPointLightColor;

let setPointLightColor = (value, component, state) =>
  PointLightAPI.setPointLightColor(component, value, state);

let getPointLightIntensity = PointLightAPI.getPointLightIntensity;

let setPointLightIntensity = (value, component, state) =>
  PointLightAPI.setPointLightIntensity(component, value, state);

let getPointLightConstant = PointLightAPI.getPointLightConstant;

let setPointLightConstant = (value, component, state) =>
  PointLightAPI.setPointLightConstant(component, value, state);

let getPointLightLinear = PointLightAPI.getPointLightLinear;

let setPointLightLinear = (value, component, state) =>
  PointLightAPI.setPointLightLinear(component, value, state);

let getPointLightQuadratic = PointLightAPI.getPointLightQuadratic;

let setPointLightQuadratic = (value, component, state) =>
  PointLightAPI.setPointLightQuadratic(component, value, state);

let getPointLightRange = PointLightAPI.getPointLightRange;

let setPointLightRange = (value, component, state) =>
  PointLightAPI.setPointLightRange(component, value, state);

let setPointLightRangeLevel = PointLightAPI.setPointLightRangeLevel;

let isMaxCount = PointLightAPI.isMaxCount;

let getCurrentCount = engineState =>
  RecordPointLightMainService.getRecord(engineState).renderLightArr
  |> CountLightService.getLightCount;

let isMaxCountByCount = (count, engineState) =>
  getCurrentCount(engineState)
  + count > BufferPointLightService.getBufferMaxCount();

let setPointLightIsRender = PointLightAPI.setPointLightIsRender;

let getBufferMaxCount = BufferPointLightService.getBufferMaxCount;

let getLightCount = engineState =>
  CountLightService.getLightCount(
    RecordPointLightMainService.getRecord(engineState).renderLightArr,
  );