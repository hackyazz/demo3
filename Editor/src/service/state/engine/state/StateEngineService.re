let deepCopyForRestore = Wonderjs.StateAPI.deepCopyForRestore;

let createState = Wonderjs.StateAPI.createState;

let createStateData = Wonderjs.StateAPI.createStateData;

let getStateFromData = Wonderjs.StateAPI.getStateFromData;

let unsafeGetState = Wonderjs.StateAPI.unsafeGetState;

let setStateToData = Wonderjs.StateAPI.setStateToData;

let setState = Wonderjs.StateAPI.setState;

let setIsDebug = Wonderjs.StateAPI.setIsDebug;