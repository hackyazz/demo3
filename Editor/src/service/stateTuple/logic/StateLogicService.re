let getEngineStateToGetData = handleFunc =>
  StateEngineService.unsafeGetState() |> handleFunc;

let getAndSetEngineState = handleFunc =>
  StateEngineService.unsafeGetState()
  |> handleFunc
  |> StateEngineService.setState
  |> ignore;

let refreshEngineState = engineState =>
  engineState
  |> DirectorEngineService.loopBody(0.)
  |> StateEngineService.setState
  |> ignore;

let refreshEngineStateAndReturnEngineState = engineState =>
  engineState |> DirectorEngineService.loopBody(0.);

let getAndRefreshEngineState = () =>
  StateEngineService.unsafeGetState()
  |> DirectorEngineService.loopBody(0.)
  |> StateEngineService.setState
  |> ignore;

let getAndRefreshEngineStateWithFunc = handleFunc =>
  StateEngineService.unsafeGetState()
  |> handleFunc
  |> DirectorEngineService.loopBody(0.)
  |> StateEngineService.setState
  |> ignore;

let loopBodyWhenStop = engineState =>
  StateEditorService.getIsRun() ?
    engineState : engineState |> DirectorEngineService.loopBody(0.);

let getAndRefreshEngineStateWhenStop = () =>
  StateEngineService.unsafeGetState()
  |> loopBodyWhenStop
  |> StateEngineService.setState
  |> ignore;

let renderWhenStop = engineState =>
  StateEditorService.getIsRun() ?
    engineState : engineState |> DirectorEngineService.loopBody(0.);

let getEditorState = handleFunc => StateEditorService.getState() |> handleFunc;

let getAndSetEditorState = handleFunc =>
  StateEditorService.getState()
  |> handleFunc
  |> StateEditorService.setState
  |> ignore;

let getStateToGetData = handleFunc =>
  (StateEditorService.getState(), StateEngineService.unsafeGetState())
  |> handleFunc;

let getAndSetState = handleFunc => {
  let (editorState, engineState) =
    (StateEditorService.getState(), StateEngineService.unsafeGetState())
    |> handleFunc;

  editorState |> StateEditorService.setState |> ignore;
  engineState |> StateEngineService.setState |> ignore;

  ();
};

let setState = ((editorState, engineState)) => {
  editorState |> StateEditorService.setState |> ignore;
  engineState |> StateEngineService.setState |> ignore;

  ();
};