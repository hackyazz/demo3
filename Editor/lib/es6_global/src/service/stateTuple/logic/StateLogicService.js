

import * as Curry from "../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as StateEditorService$WonderEditor from "../../state/editor/StateEditorService.js";
import * as StateEngineService$WonderEditor from "../../state/engine/state/StateEngineService.js";
import * as DirectorEngineService$WonderEditor from "../../state/engine/DirectorEngineService.js";

function getEngineStateToGetData(handleFunc) {
  return Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0));
}

function getAndSetEngineState(handleFunc) {
  StateEngineService$WonderEditor.setState(Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function refreshEngineState(engineState) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, engineState));
  return /* () */0;
}

function refreshEngineStateAndReturnEngineState(engineState) {
  return DirectorEngineService$WonderEditor.loopBody(0, engineState);
}

function getAndRefreshEngineState(param) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function getAndRefreshEngineStateWithFunc(handleFunc) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0))));
  return /* () */0;
}

function loopBodyWhenStop(engineState) {
  var match = StateEditorService$WonderEditor.getIsRun(/* () */0);
  if (match) {
    return engineState;
  } else {
    return DirectorEngineService$WonderEditor.loopBody(0, engineState);
  }
}

function getAndRefreshEngineStateWhenStop(param) {
  StateEngineService$WonderEditor.setState(loopBodyWhenStop(StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function renderWhenStop(engineState) {
  var match = StateEditorService$WonderEditor.getIsRun(/* () */0);
  if (match) {
    return engineState;
  } else {
    return DirectorEngineService$WonderEditor.loopBody(0, engineState);
  }
}

function getEditorState(handleFunc) {
  return Curry._1(handleFunc, StateEditorService$WonderEditor.getState(/* () */0));
}

function getAndSetEditorState(handleFunc) {
  StateEditorService$WonderEditor.setState(Curry._1(handleFunc, StateEditorService$WonderEditor.getState(/* () */0)));
  return /* () */0;
}

function getStateToGetData(handleFunc) {
  return Curry._1(handleFunc, /* tuple */[
              StateEditorService$WonderEditor.getState(/* () */0),
              StateEngineService$WonderEditor.unsafeGetState(/* () */0)
            ]);
}

function getAndSetState(handleFunc) {
  var match = Curry._1(handleFunc, /* tuple */[
        StateEditorService$WonderEditor.getState(/* () */0),
        StateEngineService$WonderEditor.unsafeGetState(/* () */0)
      ]);
  StateEditorService$WonderEditor.setState(match[0]);
  StateEngineService$WonderEditor.setState(match[1]);
  return /* () */0;
}

function setState(param) {
  StateEditorService$WonderEditor.setState(param[0]);
  StateEngineService$WonderEditor.setState(param[1]);
  return /* () */0;
}

export {
  getEngineStateToGetData ,
  getAndSetEngineState ,
  refreshEngineState ,
  refreshEngineStateAndReturnEngineState ,
  getAndRefreshEngineState ,
  getAndRefreshEngineStateWithFunc ,
  loopBodyWhenStop ,
  getAndRefreshEngineStateWhenStop ,
  renderWhenStop ,
  getEditorState ,
  getAndSetEditorState ,
  getStateToGetData ,
  getAndSetState ,
  setState ,
  
}
/* StateEditorService-WonderEditor Not a pure module */
