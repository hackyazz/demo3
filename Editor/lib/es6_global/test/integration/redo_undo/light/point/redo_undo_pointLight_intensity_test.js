

import * as Curry from "../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as RedoUndoTool$WonderEditor from "../../tool/RedoUndoTool.js";
import * as GameObjectTool$WonderEditor from "../../../../tool/GameObjectTool.js";
import * as EventListenerTool$WonderEditor from "../../../../unit/tool/EventListenerTool.js";
import * as StateLogicService$WonderEditor from "../../../../../src/service/stateTuple/logic/StateLogicService.js";
import * as DirectorToolEngine$WonderEditor from "../../../../tool/engine/DirectorToolEngine.js";
import * as StateEngineService$WonderEditor from "../../../../../src/service/state/engine/StateEngineService.js";
import * as MainEditorLightTool$WonderEditor from "../../../inspector/composable_component/sceneTree_inspector/light/tool/MainEditorLightTool.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../tool/MainEditorSceneTool.js";
import * as PointLightEngineService$WonderEditor from "../../../../../src/service/state/engine/PointLightEngineService.js";
import * as MainEditorPointLightTool$WonderEditor from "../../../inspector/composable_component/sceneTree_inspector/light/point/tool/MainEditorPointLightTool.js";
import * as BuildComponentForCurryTool$WonderEditor from "../../../../tool/BuildComponentForCurryTool.js";
import * as NoWorkerJobConfigToolEngine$WonderEditor from "../../../../tool/engine/NoWorkerJobConfigToolEngine.js";

describe("redo_undo: pointLight intensity", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        var _changeIntensity = function (value) {
          var light = GameObjectTool$WonderEditor.getCurrentGameObjectPointLightComponent(/* () */0);
          return MainEditorPointLightTool$WonderEditor.changeIntensityAndBlur(light, PointLightEngineService$WonderEditor.getPointLightIntensity(light, StateEngineService$WonderEditor.unsafeGetState(/* () */0)), value, undefined, undefined, /* () */0);
        };
        var _simulateTwiceChangeIntensity = function () {
          MainEditorLightTool$WonderEditor.setLightTypeToBePointLight(/* () */0);
          _changeIntensity(10.1);
          return _changeIntensity(12.12);
        };
        var _beforeEach = function () {
          MainEditorSceneTool$WonderEditor.initStateWithJob(sandbox, NoWorkerJobConfigToolEngine$WonderEditor.buildNoWorkerJobConfig(undefined, undefined, undefined, undefined, undefined, /* () */0), undefined, undefined, undefined, /* () */0);
          Curry._1(EventListenerTool$WonderEditor.stubGetElementByIdReturnFakeDom, Curry._1(EventListenerTool$WonderEditor.buildFakeDom, /* () */0));
          MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, (function () {
                  return GameObjectTool$WonderEditor.setCurrentSceneTreeNode(StateLogicService$WonderEditor.getEngineStateToGetData(MainEditorSceneTool$WonderEditor.getDirectionLightInDefaultScene));
                }));
          return DirectorToolEngine$WonderEditor.prepareAndInitAllEnginState(/* () */0);
        };
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                return /* () */0;
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        return RedoUndoTool$WonderEditor.testRedoUndoTwoStep(sandbox, "prepare first step: set currentSceneTreeNode", /* tuple */[
                    _simulateTwiceChangeIntensity,
                    _beforeEach,
                    (function () {
                        return /* () */0;
                      })
                  ], BuildComponentForCurryTool$WonderEditor.buildPointLight);
      }));

export {
  
}
/*  Not a pure module */
