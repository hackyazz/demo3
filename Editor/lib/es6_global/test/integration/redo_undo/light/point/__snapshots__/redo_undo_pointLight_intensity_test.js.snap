// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: pointLight intensity prepare first step: set currentSceneTreeNode test redo operate test redo one step undo step which from second to zero, redo step which from zero to first 1`] = `
<article
  className="wonder-point-light"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Color
    </div>
    <div
      className="item-content item-color"
    >
      <div
        className="color-hex"
        onClick={[Function]}
        style={
          Object {
            "background": "#ffffff",
          }
        }
      />
      <div
        className="color-select"
        onClick={[Function]}
      >
        <img
          src="./public/img/color.png"
        />
      </div>
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Intensity
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="10.1"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Constant
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="1"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Linear
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="0.07"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Quadratic
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="0.017"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Range
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="65"
      />
    </div>
  </article>
</article>
`;

exports[`redo_undo: pointLight intensity prepare first step: set currentSceneTreeNode test redo operate test redo two step undo step which from second to zero,redo step which from zero to second 1`] = `
<article
  className="wonder-point-light"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Color
    </div>
    <div
      className="item-content item-color"
    >
      <div
        className="color-hex"
        onClick={[Function]}
        style={
          Object {
            "background": "#ffffff",
          }
        }
      />
      <div
        className="color-select"
        onClick={[Function]}
      >
        <img
          src="./public/img/color.png"
        />
      </div>
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Intensity
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="12.12"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Constant
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="1"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Linear
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="0.07"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Quadratic
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="0.017"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Range
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="65"
      />
    </div>
  </article>
</article>
`;

exports[`redo_undo: pointLight intensity prepare first step: set currentSceneTreeNode test undo operate test not undo 1`] = `
<article
  className="wonder-point-light"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Color
    </div>
    <div
      className="item-content item-color"
    >
      <div
        className="color-hex"
        onClick={[Function]}
        style={
          Object {
            "background": "#ffffff",
          }
        }
      />
      <div
        className="color-select"
        onClick={[Function]}
      >
        <img
          src="./public/img/color.png"
        />
      </div>
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Intensity
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="12.12"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Constant
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="1"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Linear
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="0.07"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Quadratic
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="0.017"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Range
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="65"
      />
    </div>
  </article>
</article>
`;

exports[`redo_undo: pointLight intensity prepare first step: set currentSceneTreeNode test undo operate test undo one step step which from second to first 1`] = `
<article
  className="wonder-point-light"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Color
    </div>
    <div
      className="item-content item-color"
    >
      <div
        className="color-hex"
        onClick={[Function]}
        style={
          Object {
            "background": "#ffffff",
          }
        }
      />
      <div
        className="color-select"
        onClick={[Function]}
      >
        <img
          src="./public/img/color.png"
        />
      </div>
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Intensity
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="10.1"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Constant
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="1"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Linear
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="0.07"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Quadratic
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="0.017"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Range
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="65"
      />
    </div>
  </article>
</article>
`;

exports[`redo_undo: pointLight intensity prepare first step: set currentSceneTreeNode test undo operate test undo two step step which from second to zero 1`] = `
<article
  className="wonder-point-light"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Color
    </div>
    <div
      className="item-content item-color"
    >
      <div
        className="color-hex"
        onClick={[Function]}
        style={
          Object {
            "background": "#ffffff",
          }
        }
      />
      <div
        className="color-select"
        onClick={[Function]}
      >
        <img
          src="./public/img/color.png"
        />
      </div>
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Intensity
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="1"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Constant
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="1"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Linear
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="0.07"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Quadratic
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="0.017"
      />
    </div>
  </article>
  <article
    className="inspector-item wonder-float-input"
  >
    <div
      className="item-header component-label"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
    >
      Range
    </div>
    <div
      className="item-content"
    >
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="65"
      />
    </div>
  </article>
</article>
`;
